### ISSUE TITLE
TabPanel improvements

### ISSUE THREAD

**Issue by kormik** [rjaros/kvision#183]
I have a couple of ideas to improve TabPanel base on our experience. We use this component quite a lot and we often need to manipulate tabs programmatically. In our use case, we use tabs like in IDE, where user can open multiple tabs with different files so we don't have a static list of tabs at the beginning. 
We need to add/remove tabs, count tabs, reorder them(ideally using drag and drop), rename tab, highlight tab, change icon, change closeable state, etc.

I would preffer more tight relation between tab nagigation component and tab content component using an interface. For example tab component should implement some interface that defines things used in navigation component:

```
interface Tab: Component {
var label
var icon
var closeable
}
```

It requires those properties to be observables. Or it can be implemented other way round when nav component is injected into tab component after adding the component into TabPanel.

I am opening this issue to think about it and discuss it.

**Comment by rjaros** [comment]
I think adding a `Tab` component is a good idea. It will make things easier even for static use cases. E.g. tab panel could be finally created with a decent DSL:
```
tabPanel {
   tab("First tab", "fa-times") {
       div("First tab content")
   }
   tab("Second tab", closable = true) {
       div("Second tab content")
   }
}
```
Also currently `TabPanel` is internally based on a `StackPanel` container. We should think whether it is still good idea.

**Comment by rjaros** [comment]
I've rewritten `TabPanel` component almost from scratch and I've added `Tab` component to fully model tabs navigation. New implementation allows to:
- define tabs with simple DSL (just like in the comment above)
- dynamically add and remove tabs
- treat tabs as normal containers (can contain more then one child component)
- dynamically modify most of tab properties (label, icon, image, closable)
- get tabs list
- get number of tabs
- get tab by index
- get index of a tab
- find tab containing a given child component
- dynamically reorder tabs
- allow user to reorder tabs with drag & drop

It should be also fully compatible with an old implementation. Please report any issues.

**Comment by rjaros** [comment]
Fixed in 3.15.0
